cpp/src/cylon/net/ops/gather.cpp

```diff
@@ -153,5 +153,70 @@ cylon::Status cylon::mpi::Gather(const std::shared_ptr<cylon::TableSerializer> &
   return cylon::Status::OK();
 }
 
+cylon::Status cylon::mpi::AllGather(const std::shared_ptr<cylon::TableSerializer> &serializer,
+                                    const std::shared_ptr<cylon::Allocator> &allocator,
+                                    std::vector<int32_t> &all_buffer_sizes,
```

Comment:
shouldn't `all_buffer_sizes` be `int64`, because it is about lengths?

Reply:
in cudf, column sizes are int32. so, we can either use int64 and cast cudf sizes to int64, or maybe we can use a template type. can we have a template type with two possible values: int32 and int64?